### ISSUE TITLE
Routing not completed

### ISSUE THREAD

**Issue by Ecureuilvirtuel** [hoaproject/Router#38]
Environnement :
windows 7
PHP 5.4.34

Problem : 
My route is not dispatche correctly, so i speak with hywan and we try to isolate the problem with hoa/router.

So the code is : 

$router = new Hoa\Router\Http();
$router->get('/supervisor', 'supervisor', 'index');
$router->route('/supervisor');

print_r(
    $router->getTheRule()
);

The expected response :

Array
(
    [0] => 0
    [1] => /supervisor
    [2] => Array
        (
            [0] => get
        )

    [3] => supervisor
    [4] => index
    [5] => 
    [6] => Array
        (
            [_uri] => supervisor
            [_method] => get
            [_domain] => 
            [_subdomain] => 
            [_call] => index
            [_able] => 
            [_request] => Array
                (
                )

        )

)

And my response is :

Array
(
    [0] => 0
    [1] => supervisor
    [2] => Array
        (
            [0] => get
        )

    [3] => /supervisor
    [4] => supervisor
    [5] => index
    [6] => Array
        (
            [_uri] => supervisor
            [_method] => get
            [_domain] => bddobs.dev
            [_subdomain] => 
            [_call] => supervisor
            [_able] => index
            [_request] => Array
                (
                )

        )

)

**Comment by Jir4** [comment]
For me `id` is missing in your syntax : 
```php
<?php
$router->get('/supervisor', 'supervisor', 'index');
````
You use the `get()` magic method with three arguments : `pattern`, `call`and `able`.

But in https://github.com/hoaproject/Router/blob/master/Generic.php#L163 and https://github.com/hoaproject/Router/blob/master/Generic.php#L108-L125 we can see that the correct syntax must be something like :

```
get(<id>,<pattern>,<call>,<able>)
```

So i think you have to try something like :
```php
<?php
$router->get('SuperVisorIndex','/supervisor','supervisor','Index')
```

I can't try for now but i hope it's the good answer.

**Comment by Ecureuilvirtuel** [comment]
It does't change anything

**Comment by Jir4** [comment]
Uhm ok, i'll make a try with your versions of `router` and `dispatcher`

**Comment by Jir4** [comment]
What did you want to do exactly ? if i understood correctly you want to call the `Index` method of your `supervisor` class when the `http://domain/supervisor` uri is called. That's it ?

**Comment by Ecureuilvirtuel** [comment]
It's that. I have un directory Application. Inside A directory Controller and Public.
In the file Public/index.php i specify a route which redirect me to the Action index of the class Controller/SupervisorController with the url http://domain/supervisor.

**Comment by Jir4** [comment]
Ok so you need some configuration on your dispatcher to tell it what it have to match :

try with this :
```php
$dispatcher = new Dispatcher\ClassMethod([
    'synchronous.call'  => 'Application\Controller\(:call:U:)'
]);
```

**Comment by Jir4** [comment]
Does it works ? @Ecureuilvirtuel

**Comment by Ecureuilvirtuel** [comment]
I have this configuration file for the dispatcher :

return array (
  'keywords' =>
  array (
    'controller' => 'main',
    'action' => 'main',
    'method' => NULL,
  ),
  'parameters' =>
  array (
    'synchronous.call' => 'Application\Controller\(:controller:U:)Controller',
    'synchronous.able' => '(:action:U:)Action',
    'asynchronous.call' => '(:%synchronous.controller:)',
    'asynchronous.able' => '(:%synchronous.action:)Async',
  ),
);

This configuration is good but the problem is still present.

**Comment by Jir4** [comment]
There is a type in your configuration, a `\` is missing before `(:controller:U)`

**Comment by Ecureuilvirtuel** [comment]
This is github which remove the "\".

**Comment by Jir4** [comment]
:( each time i think i have found the answer ^^

Here my test :
![ecureuil](https://cloud.githubusercontent.com/assets/717155/12719005/b5031134-c8f2-11e5-9bf6-768b5f1f1a3b.png)

For me everything is ok !

**Comment by Ecureuilvirtuel** [comment]
Good job, problem solved :

my config was wrong : 

    'synchronous.call' => 'Application\Controller\(:controller:U:)Controller',
    'synchronous.able' => '(:action:U:)Action',

and the solution is :

    'synchronous.call' => 'Application\Controller\(:call:U:)Controller',
    'synchronous.able' => '(:able:U:)Action',

Thank's a lot :+1:

**Comment by Hywan** [comment]
@Ecureuilvirtuel Can  you close the issue please?
@Jir4 Good job!

**Comment by Jir4** [comment]
:bowtie: :sweat_smile:

**Comment by Ecureuilvirtuel** [comment]
closes #38

**Comment by Ecureuilvirtuel** [comment]
@Hywan I can't close the issue, I have no right ...
